aws-s3-transfer-manager/tests/e2e_transfer_test.rs

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// #![cfg(e2e)]
+
+// Tests here requires AWS account with pre-configured S3 bucket to run the tests.
+// Refer to https://github.com/awslabs/aws-c-s3/tree/main/tests/test_helper to help set up the S3 in the account
+// Set CRT_S3_TEST_BUCKET_NAME environment variables to the bucket created.
+// By default, it uses s3-tm-rs-test-bucket
+
+mod test_utils;
+
+use core::num;
+
+use aws_s3_transfer_manager::io::InputStream;
+use aws_s3_transfer_manager::metrics::unit::ByteUnit;
+use aws_s3_transfer_manager::operation::upload::ChecksumStrategy;
+use aws_sdk_s3::types::ChecksumMode;
+use test_utils::drain;
+
+use crate::test_utils::setup_tracing;
```

Comment:
This looks unused but also we have a utility for this already:

```rs
use aws_smithy_runtime::test_util::capture_test_logs::show_test_logs;

#[tokio::test]
async fn my_test() {
    let _logs = show_test_logs();
    ...
}
```

`show_test_logs` will register a subscriber that works as expected until the guard returned is dropped.

Also as an FYI we have [capture_test_logs](https://github.com/smithy-lang/smithy-rs/blob/main/rust-runtime/aws-smithy-runtime/src/test_util/capture_test_logs.rs#L47) which can be used to capture log output and make assertions on log output.